skrifa/src/meta/metrics.rs

```diff
@@ -0,0 +1,434 @@
+        if !used_typo_metrics {
+            if let Ok(hhea) = font.hhea() {
+                metrics.ascent = hhea.ascender().to_i16() as f32 * scale;
+                metrics.descent = hhea.descender().to_i16() as f32 * scale;
+                metrics.leading = hhea.line_gap().to_i16() as f32 * scale;
+            }
+            if metrics.ascent == 0.0 && metrics.descent == 0.0 {
+                if let Some(os2) = &os2 {
+                    if os2.s_typo_ascender() != 0 || os2.s_typo_descender() != 0 {
+                        metrics.ascent = os2.s_typo_ascender() as f32 * scale;
+                        metrics.descent = os2.s_typo_descender() as f32 * scale;
+                        metrics.leading = os2.s_typo_line_gap() as f32 * scale;
+                    } else {
+                        metrics.ascent = os2.us_win_ascent() as f32 * scale;
+                        // Win descent is always positive while other descent values are negative. Negate it
+                        // to ensure we return consistent metrics.
+                        metrics.descent = -(os2.us_win_descent() as f32 * scale);
+                    }
+                }
+            }
+        }
+        if let (Ok(mvar), true) = (font.mvar(), !coords.is_empty()) {
+            use read_fonts::tables::mvar::tags::*;
+            macro_rules! metric_delta {
+                ($tag: ident) => {
+                    mvar.metric_delta($tag, coords).unwrap_or_default().to_f64() as f32 * scale
+                };
+            }
+            metrics.ascent += metric_delta!(HASC);
+            metrics.descent += metric_delta!(HDSC);
+            metrics.leading += metric_delta!(HLGP);
+            if let Some(cap_height) = &mut metrics.cap_height {
+                *cap_height += metric_delta!(CPHT);
+            }
+            if let Some(x_height) = &mut metrics.x_height {
+                *x_height += metric_delta!(XHGT);
+            }
+            if let Some(underline) = &mut metrics.underline {
+                underline.offset += metric_delta!(UNDO);
+                underline.thickness += metric_delta!(UNDS);
+            }
+            if let Some(strikeout) = &mut metrics.strikeout {
+                strikeout.offset += metric_delta!(STRO);
+                strikeout.thickness += metric_delta!(STRS);
+            }
+        }
+        metrics
+    }
+}
+
+/// Glyph specific metrics.
+#[derive(Clone)]
+pub struct GlyphMetrics<'a> {
+    glyph_count: u16,
+    scale: f32,
+    h_metrics: &'a [LongMetric],
+    default_advance_width: u16,
+    lsbs: &'a [BigEndian<i16>],
+    hvar: Option<Hvar<'a>>,
+    loca_glyf: Option<(Loca<'a>, Glyf<'a>)>,
+    coords: &'a [NormalizedCoord],
+}
+
+impl<'a> GlyphMetrics<'a> {
+    /// Creates new glyph metrics from the given font, size in pixels per em units, and normalized
+    /// variation coordinates.
+    ///
+    /// If `size` is `None` or `Some(0.0)`, resulting metric values will be in font units.
+    pub fn from_font(
```

Comment:
this might just be me but ... from_font makes me think there is some other option, like from_bytes vs from_file or some such. Here I don't know that you'd ever be able to create them for anything but a font so I wonder why this isn't `new`?

Reply:
Not sure if I anticipated other constructors. Having a single one named new makes sense.